src/pages/developing/hardware-back-button.md

```diff
@@ -0,0 +1,202 @@
+  ev.detail.register(10, (processNextHandler) => {
+    console.log('Handler was called!');
+    
+    processNextHandler();
+  });
+});
+```
+</docs-tab>
+</docs-tabs>
+
+This example shows how to indicate to Ionic Framework that you want the next handler to be fired. All callbacks are provided with a `processNextHandler` function as a parameter. Calling this will cause the next handler, if any exists, to be fired.
+
+## Handlers with the Same Priorities
+
+Internally, Ionic Framework uses something similar to a priority queue to manage hardware back button handlers. The handler with the largest priority value will be called first. In the event that there are multiple handlers with the same priority value, the _last_ handler of the same priority added to this queue will be the first handler to be called.
+
+```javascript
+document.addEventListener('ionBackButton', (ev) => {
+  ev.detail.register(10, (processNextHandler) => {
+    console.log('Handler A was called!');
+    
+    processNextHandler();
+  });
+  
+  ev.detail.register(10, (processNextHandler) => {
+    console.log('Handler B was called!');
+    
+    processNextHandler();
+  });
+});
+```
+
+In the example above, both handlers A and B have a priority of 10. Since handler B was registered last, Ionic Framework will call handler B before it calls handler A.
+
+
+## Exiting the App
+
+In some scenarios, it may be desirable to quit the app when pressing the hardware back button. This can be achieved through the use of the `ionBackButton` event and methods that Capacitor/Cordova provide.
+
+<docs-tabs>
+<docs-tab tab="javascript">
+
+```typescript
+import { Plugins } from '@capacitor/core';
+import { BackButtonEvent } from '@ionic/core';
+const { App } = Plugins;
+
+...
+
+const routerEl = document.querySelector('ion-router');
+document.addEventListener('ionBackButton', (ev: BackButtonEvent) => {
+  ev.detail.register(-1, () => {
+    const path = window.location.pathname;
+    if (path === routerEl.root) {
+      App.exitApp();
+    }
+  });
+});
+```
+</docs-tab>
+<docs-tab tab="angular">
+
+```typescript
+import { IonRouterOutlet, Platform } from '@ionic/angular';
+import { Plugins } from '@capacitor/core';
+const { App } = Plugins;
+
+...
+
+constructor(
+  private platform: Platform,
+  private routerOutlet: IonRouterOutlet
+) {
+  this.platform.backButton.subscribeWithPriority(-1, () => {
+    if (!this.routerOutlet.canGoBack()) {
+      App.exitApp();
+    }
+  });
+}
+
+```
+</docs-tab>
+</docs-tabs>
+
+This examples shows the application exiting when the user presses the hardware back button and there is nothing left in the navigation stack. It is also possible to display a quit confirmation dialog before quitting the app.
+
+It is recommended to check whether or not the user is on the root page prior to exiting the application. Developers can use the `canGoBack` method on `IonRouterOutlet` in Ionic Angular. Support for this will be coming to Ionic React soon.
```

Comment:
Can we wait until we have this supported in React? We should have a React example